### PR TITLE
Feat 44: adds feature to correct NP-opto electrode positions

### LINKED ISSUE #44: Correct for NP-opto electrode locations before compression
**Is your feature request related to a problem? Please describe.**
The OpenEphys GUI does not handle correctly the geometry of NP-opto probes when saving the settings.xml, and saves the NP-opto as having the NP1.0 configuration. 
Since we automatically read the probe information from the settings file, this can lead to errors for downstream analysis.

**Describe the solution you'd like**
Before compressing and clipping, we should check for opto probes (easy from the settings file) and correct the electrode locations, so that the correct probe configuration is loaded in SpikeInterface.

**Describe alternatives you've considered**
An alternative could be to correct the probe geometry a posteriori, but since we plan to trigger the computational pipeline as soon as a new data asset is created, this is not possible.

### PR DESCRIPTION
**Pull request by alejoe91**
Closes #44

### CODE REVIEW

**Comment by alejoe91** [comment]
@alakunina this PR "corrects" the settings file for the OPTO probes.

Here is an example of the corrected config at the tip (as loaded by SpikeInterface):
image



And settings.xml (I had to zip it):
[settings.zip](https://github.com/AllenNeuralDynamics/nd-data-transfer/files/9754175/settings.zip)

File: conf/ephys_upload_job_configs.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ endpoints:
   gcp_prefix:
   codeocean_domain: https://codeocean.allenneuraldynamics.org
 jobs: # Select which jobs to run
+  correct_opto: true
```
Comment (dyf) [inline conf/ephys_upload_job_configs.yml:11]: @alejoe91 is the idea that we would just turn this flag off once OpenEphys starts saving correct locations?

Comment (alejoe91) [reply conf/ephys_upload_job_configs.yml:11]: Yes that's the idea!

**Comment by alejoe91** [comment]
@dyf now correction is done depending on the Neuropix-PXI plugin version in the settings

**Comment by dyf** [comment]
From conversation - @alejoe91 is going to remove this flag and check OpenEphys version numbers to know whether or not this needs to happen.

**Comment by dyf** [comment]
@alejoe91 should we close this now?

**Comment by alejoe91** [comment]
> From conversation - @alejoe91 is going to remove this flag and check OpenEphys version numbers to know whether or not this needs to happen.

This is implemented! https://github.com/AllenNeuralDynamics/nd-data-transfer/blob/feat-44-np-opto-correction/src/transfer/util/npopto_correction.py#L44-L49

It's ready to be merged on my side

**Pull request merged**